lib/config/browser-options.js

```diff
@@ -138,6 +138,20 @@ function buildBrowserOptions(defaultFactory, extra) {
                 if (!_.includes(['fullpage', 'viewport', 'auto'], value)) {
                     throw new Error('"screenshotMode" must be "fullpage", "viewport" or "auto"');
                 }
+            },
+            map: (value) => {
```

Comment:
This didn't come out as clean as we expected -- I had to pass original config options object as an argument `originalOptions` above. Couldn't figure out how to access `platformName` or `browserName` otherwise.

Here, in `map(value, config, currentNode)`, expression `config === currentNode` turns out to be `true`. It means that inside the `map` function we only have access to the whole config, but we can't tell which browser we're currently dealing with.

Perhaps we should move this to somewhere where it's easier to access `platformName`. For instance, [here](https://github.com/gemini-testing/hermione/blob/2692cb6c35a1c1d65bd015263ecfe287e36b4538/lib/config/index.js#L57)

Reply:
It looks like that there is a bug inside gemini-configparser. I suggest to fix it in order to don't stumble about it in the future.

It seems that enough to fix building a locator name here - https://github.com/gemini-testing/configparser/blob/master/lib/locator.js#L21. In case when `namePrefix` is empty we don't set a point before `subKey`. Then everything should work here - https://github.com/gemini-testing/configparser/blob/master/lib/core.js#L18 and `currNode` will be defined correctly.